notificationapi_android_sdk/src/main/java/com/notificationapi/notificationapi_android_sdk/activity/NotificationApiActivity.kt

```diff
@@ -0,0 +1,29 @@
+package com.notificationapi.notificationapi_android_sdk.activity
+
+import android.content.pm.PackageManager
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.notificationapi.notificationapi_android_sdk.NotificationApi
+
+open class NotificationApiActivity: AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        NotificationApi.initialize(context = this)
+    }
+
+    override fun onRequestPermissionsResult(
```

Comment:
This gets invoked after the user requests notification permissions. This will trigger a handy `onNotificationRequestPermissionResult()` function with the result so the implementing developer can just use that.

However I should not that there is a possiblity that the developer does another permission request with the _same request code_ as `NotificationApi.NOTIFICATION_PERMISSION_REQUEST`. This may cause issues, but can be fixed by changing the request code.